yt/utilities/parallel_tools/parallel_analysis_interface.py

```diff
@@ -1114,13 +1150,17 @@ def send_array(self, arr, dest, tag=0):
                 unit_metadata += ("YTArray",)
         else:
             unit_metadata = ()
-        self.comm.send((arr.dtype.str, arr.shape) + unit_metadata, dest=dest, tag=tag)
-        self.comm.Send([arr, MPI.CHAR], dest=dest, tag=tag)
+        self.comm.send(
+            (arr.dtype.str, arr.shape, transport_dtype) + unit_metadata,
+            dest=dest,
+            tag=tag,
+        )
+        self.comm.Send([arr, mpi_type], dest=dest, tag=tag)
```

Comment:
Isn't there a bug here? I think this should be
```suggestion
        self.comm.send(
            (arr.dtype.str, arr.shape, transport_dtype) + unit_metadata,
            dest=dest,
            tag=tag,
        )
        self.comm.Send([tmp, mpi_type], dest=dest, tag=tag)
```

Reply:
I think it should be `tmp`, yeah, but since it has been working I think it's possible that mpi4py was fixing it implicitly.